Infoblox Cloud Release 2.0.0

crestdatasystems opened the pull request:
<!-- REMINDER: THIS IS A PUBLIC REPO DO NOT POST HERE SECRETS/SENSITIVE DATA -->
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - ()

## Description

#### Updated Integration - Infoblox Threat Defense with DDI

- Added support for **Fetch incidents** to fetch the Security Insights or DNS Security Events as XSOAR incidents.
- Added support for **infobloxcloud-soc-insight-list** command that list SOC insights from Infoblox Cloud.
- Added support for **infobloxcloud-soc-insight-event-list** command that list events for a specific SOC Insight.
- Added support for **infobloxcloud-soc-insight-indicator-list** command that list indicators for a specific SOC Insight.
- Added support for **infobloxcloud-soc-insight-comment-list** command that list comments for a specific SOC Insight.
- Added support for **infobloxcloud-soc-insight-asset-list** command that list assets for a specific SOC Insight.
- Added support for **infobloxcloud-block-ip** command to add specified IP addresses for the provided block list.
- Added support for **infobloxcloud-unblock-ip** command to add specified IP addresses for the provided allow list.
- Added support for **infobloxcloud-block-domain** command to add specified domains for the provided block list.
- Added support for **infobloxcloud-unblock-domain** command to add specified domains for the provided allow list.
- Added support for **infobloxcloud-customlist-indicator-remove** command to remove specified indicators from the provided custom list.
- Added support for **ip** command that gets the comprehensive IP reputation and threat intelligence from Infoblox Threat Defense, including threat indicators, IPAM address information, and standard IP reputation data.
- Added support for **domain** command that gets the comprehensive domain/host reputation and threat intelligence from Infoblox Threat Defense, including threat indicators, IPAM address information and standard domain reputation data.
- Added support for **url** command that gets the comprehensive URL reputation and threat intelligence from Infoblox Threat Defense, including threat indicators, and standard URL reputation data.
- Added support for **infobloxcloud-mac-enrich** command that enriches a MAC address with DHCP lease information.

#### Added following set of Playbooks

##### Domain Enrichment - Infoblox Cloud

- This playbook enriches domains or hosts with the dossier, TIDE and asset data using Infoblox Threat Defense with DDI integration.

##### Block Indicator - Infoblox Cloud

- This playbook blocks the given IP or domain by adding it to the given block type custom list of the Infoblox Cloud platform. If prompted it also removes the provided indicators from given allow list.

##### Unblock Indicator - Infoblox Cloud

- This playbook unblocks the given IP or domain by adding it to the given allow type custom list of the Infoblox Cloud platform.

##### Indicator Enrichment - Infoblox Cloud

- This playbook enriches IP addresses, MAC addresses, domains and URLs with the dossier, DHCP lease, TIDE and asset data using Infoblox Threat Defense with DDI integration.

##### Incident Response - Infoblox Cloud

- This playbook is used to initiate the incident response. This playbook runs when an incident is selected for investigation. It will change the state from pending to active and it will list the available indicators, events, assets, and comments from Infoblox corresponding to the incident. If incident severity is found to be higher than or equivalent to medium, it will create a ServiceNow incident otherwise the given incident will be assigned to an analyst.

##### MAC Enrichment - Infoblox Cloud

- This playbook enriches MAC addresses with DHCP lease information using Infoblox Threat Defense with DDI integration.

##### URL Enrichment - Infoblox Cloud

- This playbook enriches URL with the dossier and TIDE data using Infoblox Threat Defense with DDI integration.

##### IP Enrichment - Infoblox Cloud

- This playbook enriches IP addresses with the dossier, TIDE and asset data using Infoblox Threat Defense with DDI integration.

## Must have
- [x] Tests
- [x] Documentation

merit-maita commented on Packs/InfobloxBloxOne/Integrations/InfobloxBloxOneThreatDefense/InfobloxBloxOneThreatDefense.yml:
```diff
@@ -106,11 +210,1758 @@ script:
+      description: The confidence level provided by the vendor for the threat type category For example, a confidence of 90 for the threat type category 'malware' means that the vendor rates that this is 90% confidence of being a malware.
+      type: String
+    # Standard DBotScore context outputs (mandatory for reputation commands)
+    - contextPath: DBotScore.Indicator
+      description: The indicator that was tested.
+      type: String
+    - contextPath: DBotScore.Type
+      description: The indicator type.
+      type: String
+    - contextPath: DBotScore.Vendor
+      description: The vendor used to calculate the score.
+      type: String
+    - contextPath: DBotScore.Score
+      description: The actual score.
+      type: Number
+    - contextPath: DBotScore.Reliability
+      description: Reliability of the source providing the intelligence data.
+      type: String
+    # InfobloxCloud.IP.Threat context outputs (threat intelligence data)
```
why are they twice under ip?

merit-maita commented on Packs/InfobloxBloxOne/Integrations/InfobloxBloxOneThreatDefense/InfobloxBloxOneThreatDefense.yml:
the new content you added has the fromversion  set to 6.10 while the integration is 6.5, please change theirs to be 6.5

merit-maita reviewed (changes requested):
thank you for your contribution, it looks great!
general notes:
please replace any  "<br>" value in the readme files to be "<br/>"
incidents fields and types prefix should be exactly the name of the pack, aka instead of Infoblox it should be InfoBloxOne, or remove it at all

merit-maita reviewed (changes requested):
Thanks for the updates, there're still a couple of points to consider
- please replace any < br > in the md files with < br/ > (no space in between, i had to add space to be able to see it)
- take a look at the failing checks in the pre-commit step
- although you filled the contribution form, a label is missing, can you please try refilling it and make sure to provide the correct partner ID

merit-maita commented:
Current Status:
the pr is on hold as they're planing to introduce a new fetch type in the upcoming week.
they'll update the PR once the changes are ready for certification.
Slack DFIR: Deep Patel

crestdatasystems commented:
> Current Status: the pr is on hold as they're planing to introduce a new fetch type in the upcoming week. they'll update the PR once the changes are ready for certification. Slack DFIR: Deep Patel

Hi @merit-maita,

We have updated the PR with added fetch mechanism for DNS Security Events. Also, we have updated the Insight related components with the added "SOC" word as a prefix.

merit-maita commented:
@Benimanela can you please take a look

merit-maita commented:
> > > * br
> > 
> > 
> > regarding this comment, please make sure to replace < br > with < br/ >
> 
> Hi @merit-maita,
> 
> We have checked the integration and description README file for the "< br >" tag and not able to find it. Can you please specify any files where it is present?
> 
> Thanks.

please remove the space between the characters to find them, they're in the md files like readme

merit-maita commented:
@crestdatasystems please handle the merge conflict to continue
